libnavigation-core/src/main/java/com/mapbox/navigation/core/routealternatives/RouteAlternativesController.kt

```diff
@@ -181,8 +182,14 @@ internal class RouteAlternativesController constructor(
                             return@processRouteAlternatives
                         }
 
+                    val updatedAlternatives = restoreChargingStationsMetadata(
```

Comment:
There are 2 possible options here:

1. NN don't fix server-added charging stations when requesting alternatives. Then we don't need to restore anything here;
2. NN do fix server-added charging stations when requesting alternatives. Then it's them who should restore them.

Right?
Or is it just to fix the onboard router ignoring the metadata?

Reply:
It should be the second. I added a comment about it in NN-854